ios/storage.md

```diff
@@ -161,317 +161,620 @@ func downloadData() {
 
 }
 ```
+## Using TransferUtility 
 
+This section explains how to implement upload and download functionality and a number of additional storage use cases.
 
-## Usage with GraphQL APIs (Complex Objects)
+Note: If you use the transfer utility multipart upload feature, take advantage of automatic cleanup features by setting up the [AbortIncompleteMultipartUpload] https://docs.aws.amazon.com/AmazonS3/latest/dev/intro-lifecycle-rules.html) action in your Amazon S3 bucket life cycle configuration.
+{: .callout .callout--info}
 
-Many times you might want to create logical objects that have more complex data, such as images or videos, as part of their structure. For example, you might create a Person type with a profile picture or a Post type that has an associated image. You can use AWS AppSync to model these as GraphQL types. If any of your mutations have a variable with bucket, key, region, mimeType, and localUri fields, the SDK uploads the file to Amazon S3 for you.
+### Transfer Utility Options
 
-Update your schema as follows to add the S3Object and S3ObjectInput types for the file, and a new mutation named CreatePostWithFileInputMutation:
+You can use the `AWSS3TransferUtilityConfiguration` object to configure the operations of the `TransferUtility`.
 
-```graphql
-input CreatePostInput {
-    author: String!
-    title: String
-    content: String
-    url: String
-    ups: Int
-    downs: Int
-    version: Int!
-}
+**isAccelerateModeEnabled**
+The isAccelerateModeEnabled option lets you to upload and download content from a bucket that has Transfer Acceleration enabled on it. See [Transfer Acceleration](https://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html) for information on how to enable transfer acceleration for your bucket.
 
-input CreatePostWithFileInput {
-    author: String!
-    title: String
-    content: String
-    url: String
-    ups: Int
-    downs: Int
-    file: S3ObjectInput!
-    version: Int!
-}
+This option is set to false by default.
 
-input DeletePostInput {
-    id: ID!
-}
+```swift
+//Setup credentials
```

Comment:
Can we put the `credentialsProvider` and `configuration` object in a separate section called "Manual Configuration" and point out that they should use the `AWSMobileClient` to manage all auth state, and link to that section in the docs directly. Then say something like "However, if you would like to manually provide your Cognito provider and Region you can do so with the following commands..."

We really want to start pointing people to use the new auth tooling with the AWSMobileClient now that it has all these features to ease on-boarding and other customer issues.

Reply:
Added a link to the authentication module.